src/node/history.h

```diff
@@ -383,6 +381,35 @@ namespace ccf
     std::optional<ResultCallbackHandler> on_result;
     std::optional<ResponseCallbackHandler> on_response;
 
+    void execute_pending(kv::Version v)
+    {
+      auto* p = pending_inserts.get_head();
+      while (p != nullptr)
+      {
+        auto* next = p->next;
+        if (p->version <= v)
+        {
+          add_result(p->id, p->version, *p->replicated);
+          pending_inserts.remove(p);
+        }
+        p = next;
+      }
+    }
+
+    void discard_pending(kv::Version v)
```

Comment:
Considering those methods are identical except for one line, I think we should consider a single flush_pending(kv::Version v, bool apply).

Reply:
renamed `execute_pending` to `flush_pending`